runtime_test.go

```diff
@@ -665,6 +665,30 @@ func TestRuntime_Close_ClosesCompiledModules(t *testing.T) {
 	}
 }
 
+func TestRuntime_Close_Instantiate_ShouldErr(t *testing.T) {
+	t.Run("TestRuntime_Close_Instantiate_ShouldErr", func(t *testing.T) {
+		engine := &mockEngine{name: "mock", cachedModules: map[*wasm.Module]struct{}{}}
+		conf := *engineLessConfig
+		conf.newEngine = func(context.Context, api.CoreFeatures, filecache.Cache) wasm.Engine { return engine }
+		r := NewRuntimeWithConfig(testCtx, &conf)
+		defer r.Close(testCtx)
+
+		// Normally compiled modules are closed when instantiated but this is never instantiated.
+		_, err := r.CompileModule(testCtx, binaryNamedZero)
+		require.NoError(t, err)
+		require.Equal(t, uint32(1), engine.CompiledModuleCount())
+
+		err = r.Close(testCtx)
+		require.NoError(t, err)
+
+		// Closing the runtime should remove the compiler cache if cache is not configured.
+		require.True(t, engine.closed)
+
+		_, err = r.InstantiateModuleFromBinary(testCtx, binaryNamedZero)
```

Comment:
add check for CompileModule too

Reply:
maybe I should just move `failIfClosed()` to `CompileModule()`, since `InstantiateModuleFromBinary()` calls that